src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -955,9 +955,13 @@ public int compareTo(CaseResult that) {
         if (this == that) {
             return 0;
         }
-        int r = this.getFullName().compareTo(that.getFullName());
-        if (r != 0) {
-            return r;
+        int r1 = this.className.compareTo(that.className);
```

Comment:
The behaviour looks to be different here, I don't know specifically how this is triggered and would need to do some research, this is now only looking at the class name and not class name and test name as before

Reply:
Thanks. The previous code compared getFullName() which is `className + '.' + getName()` and it is a lexicographical comparison. So when the class name was different, it would in fact not have looked at the rest of the string including the test name. This makes it the same behavior as after the change.

Reply:
Ah I missed that it does that check further down if the class name matches